src/kernel/dbwrapper_options.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_KERNEL_DBWRAPPER_OPTIONS_H
+#define BITCOIN_KERNEL_DBWRAPPER_OPTIONS_H
+
+#include <fs.h>
+
+#include <cstddef>
+
+namespace kernel {
+
+struct DBWrapperOpts {
+    fs::path db_path;
+    size_t cache_size;
+    bool in_memory = false;
+    bool wipe_existing = false;
+    bool obfuscate_data = false;
+    bool do_compact = false;
+};
```

Comment:
You can see the other change I suggested here https://github.com/bitcoin/bitcoin/pull/25623#pullrequestreview-1045680511, but I think it simplifies and clarifies code to the do_compact option to a separate struct, as it's a low-level performance tweak that is passed directly from the user to the low-level DB code, unlike the other parameters which are under control of the high level application and indexing code.

It simplifies code by avoiding the need to have 3 near-duplicate structures DBWrapperOpts BlockTreeDBOpts CoinsViewDBOpts that require boilerplate copying individual fields between them. Instead you can 2 structure, with no duplicate fields, that just pass data directly where it needs to go.